src/app/core/common/hide-overflow-list.directive.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Directive,
+  ElementRef,
+  Input, OnDestroy, OnInit,
+  Renderer2
+} from '@angular/core';
+import { Subscription, timer } from 'rxjs';
+import { distinctUntilChanged, map } from 'rxjs/operators';
+
+@Directive({
+  selector: '[algHideOverflowList]',
+})
+export class HideOverflowListDirective implements OnInit, OnDestroy {
+  @Input() algHideOverflowListStyleClass?: string;
+  @Input() algHideOverflowListTarget?: string;
+
+  private subscription?: Subscription;
+  private nextElIdxForHide = 0;
+
+  constructor(private el: ElementRef<HTMLElement>, private renderer: Renderer2) {
+  }
+
+  ngOnInit(): void {
+    this.subscription = timer(0, 100).pipe(
+      map(() => this.el.nativeElement.scrollWidth),
+      distinctUntilChanged(),
+    ).subscribe(() => {
+      this.resetOverflowStylesAndSettings();
+      this.checkAndHideOverflowedElements();
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.subscription?.unsubscribe();
+  }
+
+  private resetOverflowStylesAndSettings(): void {
+    if (!this.algHideOverflowListStyleClass) {
+      return;
+    }
+
+    this.nextElIdxForHide = 0;
+
+    const hiddenElements: NodeListOf<Element> = this.el.nativeElement.querySelectorAll('.hidden');
+
+    hiddenElements.forEach(el => {
+      this.renderer.removeClass(el, 'hidden');
+    });
+
+    this.renderer.removeClass(this.el.nativeElement, this.algHideOverflowListStyleClass);
+  }
+
+  private checkAndHideOverflowedElements(): void {
+    if (!this.algHideOverflowListTarget || !this.algHideOverflowListStyleClass) {
+      return;
+    }
+
+    const { clientWidth, scrollWidth } = this.el.nativeElement;
+    const targetElements: NodeListOf<Element> = this.el.nativeElement.querySelectorAll(this.algHideOverflowListTarget);
+    const hiddenElements: NodeListOf<Element> = this.el.nativeElement.querySelectorAll('.hidden');
+    let checkAgain = false;
+
+    if (scrollWidth > clientWidth) {
+      targetElements.forEach((el, index) => {
+        if (index === this.nextElIdxForHide && index !== targetElements.length - 1) {
+          this.renderer.addClass(el, 'hidden');
+          checkAgain = true;
+        }
+      });
+
+      if (hiddenElements.length !== targetElements.length - 1 && checkAgain) {
+        this.nextElIdxForHide += 1;
+      }
+
+      this.renderer.addClass(this.el.nativeElement, this.algHideOverflowListStyleClass);
+    }
+
+    if (checkAgain) {
+      this.checkAndHideOverflowedElements();
```

Comment:
Recursive call with no clear end-condition. :-/
If really needed, it should be a loop (not recursive) with either a provable exit condition, or a way to leave the loop after a max number of calls.
You use the instance variable `nextElIdxForHide` for storing a state of your processing ? :-/ If it is something related to one run of your processing, it should not be an instance variable.

Reply:
Why recursive, imagine we have breadcrumb:

Item 1 -> Item 2 -> Item 3 -> Item 4 -> Item 5.

So when breadcrumb container properties (we are interest scrollWidth*) are changed it runs checkAndHideOverflowedElements method.

checkAndHideOverflowedElements method hides first item when we have overflow (when scroll inside breadcrumb) and asking checkAgain this condition, like if it's enough for hide scroll or no. If there is still scroll - hide next one item.

*Scroll means text was overflowed

Reply:
So probably it should be a loop, but not a recursion.